api/api.go

```diff
@@ -210,14 +151,51 @@ func (a *API) populateContext(ctx context.Context, w http.ResponseWriter, r *htt
 		"path":   r.URL.Path,
 	})
 
-	ctx = withRequestID(ctx, id)
-	ctx = withLogger(ctx, log)
-	ctx = withConfig(ctx, a.config)
-	ctx = withStartTime(ctx, time.Now())
-	ctx = withPayer(ctx, PaypalChargerType, &paypalProvider{a.paypal})
-	ctx = withPayer(ctx, StripeChargerType, &stripeProvider{})
-	ctx = withCoupons(ctx, a.config)
+	ctx = gcontext.WithRequestID(ctx, id)
+	ctx = gcontext.WithLogger(ctx, log)
+	ctx = gcontext.WithStartTime(ctx, time.Now())
+
+	instanceID := r.Header.Get("x-nf-id")
+	if instanceID != "" {
+		// TODO populate config
+		// env := r.Header.Get("x-nf-env")
+		// var config *conf.Configuration
+		// var err error
+		// ctx, err = withTenantConfig(ctx, config)
+		// if err != nil {
+		// 	internalServerError(w, err.Error())
+		// 	return nil
+		// }
+	}
+
+	// safety check in case of multi-tenant but missing X-NF-* headers
+	if gcontext.GetPaymentProvider(ctx) == nil {
+		internalServerError(w, "No payment provider configured")
+		return nil
+	}
 
 	log.Info("request started")
 	return ctx
 }
+
+func withTenantConfig(ctx context.Context, config *conf.Configuration) (context.Context, error) {
```

Comment:
hmmm you're right. Nevermind disregard. In the multiworld this will have to be like this. I think that we can put this as a normal middleware though in that case. Let's address that when we write the multi code